server/protocol/qdatastreamprotocol.py

```diff
@@ -127,31 +126,60 @@ def pack_message(*args: str) -> bytes:
 
     def close(self):
         """
-        Close writer stream
+        Close writer stream as soon as the buffer has emptied.
         :return:
         """
         self.writer.close()
 
+    def abort(self):
+        """
+        Close writer stream immediately discarding the buffer contents
+        :return:
+        """
+        self.writer.transport.abort()
```

Comment:
Do I understand right, that HERE we don't care about anything, we want just close the writer?

But I don't understand why? I see the catch on timeout, but why do we have to call internal transport function on writer to force abort the connection?

Reply:
Yes so this is called when we've detected unresponsive clients. We don't want to call `close` because that will still wait for the buffer to drain (which we expect isn't going to happen anytime soon in this case). Sadly the stream API does not expose `abort` directly, so we have to call it on the transport object manually.